examples/cis3-nft-sponsored-txs/src/lib.rs

```diff
@@ -1304,6 +1355,29 @@ fn contract_set_implementor<S: HasStateApi>(
     Ok(())
 }
 
+// Helper function to decode a hex string into bytes.
+#[allow(dead_code)]
+fn decode_hex(s: &str) -> Result<Vec<u8>, CustomContractError> {
+    if s.len() % 2 != 0 {
+        Err(CustomContractError::ParseError)
+    } else {
+        (0..s.len())
+            .step_by(2)
+            .map(|i| u8::from_str_radix(&s[i..i + 2], 16).map_err(|e| e.into()))
+            .collect()
+    }
+}
```

Comment:
Right, so the idea was to add a `FromStr` implementation to all these types https://docs.rs/concordium-std/latest/concordium_std/struct.HashSha2256.html https://docs.rs/concordium-std/latest/concordium_std/struct.PublicKeyEd25519.html https://docs.rs/concordium-std/latest/concordium_std/struct.PublicKeyEcdsaSecp256k1.html and the signature types.

In concordium-std directly.